spec/requests/proxying_spec.rb

```diff
@@ -196,10 +190,14 @@
     end
 
     it "sets the appropriate environment as the cookie domain" do
-      ENV["GOVUK_APP_DOMAIN_EXTERNAL"] = "integration.publishing.service.gov.uk"
-      get upstream_path
-      expect(response.headers["Set-Cookie"]).to match("domain=.integration.publishing.service.gov.uk")
-      ENV.delete("GOVUK_APP_DOMAIN_EXTERNAL")
+      assert_cookie_domain
     end
   end
+
+  def assert_cookie_domain
+    ENV["GOVUK_APP_DOMAIN_EXTERNAL"] = "integration.publishing.service.gov.uk"
+    get upstream_path
+    expect(response.headers["Set-Cookie"]).to match("domain=.integration.publishing.service.gov.uk")
+    ENV.delete("GOVUK_APP_DOMAIN_EXTERNAL")
+  end
```

Comment:
This might be more appropriate as a shared example: https://relishapp.com/rspec/rspec-core/v/3-5/docs/example-groups/shared-examples#shared-examples-are-nestable-by-context

Reply:
Haven't come across that syntax before - nice!

Reply:
This style doesn't read very well though:

`it_behaves_like "sets cookie domain properly"`

And this feels too vague:

`it_behaves_like "a proper cookie"`

Any suggestions?

Reply:
You can use `include_examples` which is an alias for `it_behaves_like` so could go for `include_examples "sets auth-bypass token cookie"`

Reply:
Also if I extract to a `shared_examples` block, I get:

```
`upstream_path` is not available on an example group (e.g. a `describe` or `context` block). It is only available from within individual examples (e.g. `it` blocks) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc).
```

Reply:
@ChrisBAshton that sounds like you might have not set up an `it` inside your shared example

Reply:
Ah, wasn't sure if you could do that - working now 👍 thanks